Program.cs

```diff
@@ -49,10 +49,11 @@ static void Main(string[] args)
 
             try
             {
-                foreach (var arg in args)
+                for (int i = 0; i < args.Length; i++)
                 {
-                    parameter = arg.Substring(0, arg.IndexOf('='));
-                    parameterValue = arg.Substring(arg.LastIndexOf('=') + 1);
+                    parameter = args[i];
+                    parameterValue = args[i + 1];
```

Comment:
For this to not pass args.Length you would need to range check before args[i + 1];